src/types.ts

```diff
@@ -1,18 +1,30 @@
-import { SetStateAction, Dispatch } from "react";
+export type DefaultItem = Record<any, any>;
 
-export type Item<T> = T & {
-  selected: boolean;
-};
+export type DefaultItemIdentifierKey = any;
+
+export interface HookArguments<T = DefaultItem, K = DefaultItemIdentifierKey> {
+  initialItems: T[];
+  itemIdentifierKey: K;
+  initialSelectedItems?: T[];
+}
 
-export interface Actions<T> {
-  toggleItem: (item: T) => void;
-  setSelectedItems: Dispatch<SetStateAction<T[]>>;
-  setItemsList: Dispatch<SetStateAction<Item<T>[]>>;
+export interface Action {
+  type: ActionType;
+  payload: Omit<Partial<HookArguments>, "itemIdentifierKey"> & {
+    item?: DefaultItem;
```

Comment:
The `item` is required when sending action of type `TOGGLE_SELECTED_STATUS`

Although here is being typed as optional because it would lead to an error when dispatching other action types that don't need the `item` data. 

There's a better way to type the action's payload according to its type and we can refactor here later on.

Reply:
Improved at [`e0afeb0` (#11)](https://github.com/LauraBeatris/use-selected-items-hook/pull/11/commits/e0afeb0cc9b4f49d5ef51f49aa54363c2cb0efd4)